### PR TITLE
Fix release workflow

### PR DESCRIPTION
**Pull request by eneskaya**
This PR finalizes the automation of our release process, with `goreleaser` it now generates

- a release draft, including all the build archives
- a push to `homebrew-komiser` to update the Formula version
- a push to docker hub including the latest, version and gitcommit tag
- a push to our `helm` repository, bumping the version

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -75,3 +77,28 @@ jobs:
...
+        env:
+          TAG: ${{ github.ref_name }}
+
+      - name: "Push changes to tailwarden/helm"
+        run: |
+          cd ./helm
+          git add .
+          git config user.name "purplin"
+          git config user.email "purplin@tailwarden.com"
+          git commit -m "Bump version to 3.0.7"
```
Comment (mlabouardy) [inline .github/workflows/release.yml:103]: the version should be dynamic not hardcoded
also, could we follow the git commit format `chore: bump version v3.0.7`

File: .goreleaser.yaml [hunk 2]
```diff
@@ -13,31 +13,63 @@ builds:
     goarch:
       - amd64
       - arm64
+    ignore:
+      - goos: windows
+        goarch: arm64
+      - goos: linux
```
Comment (mlabouardy) [inline .goreleaser.yaml:19]: why arm64 for linux is skipped?

Comment (eneskaya) [reply .goreleaser.yaml:19]: I oriented myself on the circleci build steps, where this os/arch wasn't build as far as I could see. Should I enable?

Comment (mlabouardy) [reply .goreleaser.yaml:19]: make sense, no all good then

File: Dockerfile [hunk 3]
```diff
@@ -7,7 +7,7 @@ MAINTAINER mlabouardy <mohamed@tailwarden.com>
 RUN echo "Running on $BUILDPLATFORM, building for $TARGETPLATFORM" > /log
 
 ENV SEGMENT_WRITE_KEY $SEGMENT_WRITE_KEY
-ENV VERSION 3.0.7
+ENV VERSION $VERSION
```
Comment (mlabouardy) [inline Dockerfile:10]: btw, you'll need to declare the VERSION as an ARG (see line 4 for reference)

Comment (eneskaya) [reply Dockerfile:10]: Thanks for the hint!

File: .goreleaser.yaml [hunk 4]
```diff
@@ -13,31 +13,65 @@ builds:
...
-      - tailwarden/komiser:{{ .FullCommit }}
+      - tailwarden/komiser:{{ .Version }}-amd64
+      - tailwarden/komiser:{{ .FullCommit }}-amd64
+    dockerfile: Dockerfile
     use: buildx
     build_flag_templates:
-      - "--platform linux/amd64,linux/arm64"
-      - "--build-arg SEGMENT_WRITE_KEY=$SEGMENT_WRITE_KEY"
+      - --platform=linux/amd64
+      - --build-arg=SEGMENT_WRITE_KEY=$SEGMENT_WRITE_KEY
```
Comment (mlabouardy) [inline .goreleaser.yaml:51]: you need to pass the VERSION here as well: `- --build-arg=VERSION={{ .Version }}`

**Review by mlabouardy — CHANGES_REQUESTED** [review]

**Review by mlabouardy — APPROVED** [review]

**Pull request merged**